### ISSUE TITLE
FR: update dropbear to a newer release and keep it up to date

### ISSUE THREAD

**Issue by anarcat** [koreader/koreader#11698]
**Does your feature request involve difficulty completing a task? Please describe.**

I'm trying to login to my kobo running a recent koreader over SSH and i'm failing at key authentication, only password works... I believe it's because we're using a really old version (v2018.76) that doesn't support ed25519 (added in https://github.com/mkj/dropbear/pull/91, in 2020).

**Describe the solution you'd like**

Dropbear should be updated.

**Describe alternatives you've considered**
Obviously, disabling password authentication *works*, but ewww... I've also been able to install @NiLuJe's things to get a proper SSH server on another kobo.

**Additional context**
How *does* this thing get updated anyways? maybe there should be some automation here?

I believe this causes issues for others as well, e.g. i think #8370 is probably related.

**Comment by NiLuJe** [comment]
> How _does_ this thing get updated anyways?

It doesn't because it requires a bunch of platform-specific patches that are a right PITA to keep in sync, and I already maintain fully-featured bundles for Kindle & Kobo, as you mentioned earlier.

(And, unfortunately, the KOReader patches are trickier than the one used there, so the work isn't 100% portable).

**Comment by NiLuJe** [comment]
As for the original issue, yes, your analysis is correct, the horribly old version being used doesn't support modern key formats.

**Comment by benoit-pierre** [comment]
> i.e., I'd be very happy to just murder the SSH plugin entirely.

That would be a shame. I use an updated dropbear version in my meson branch (2022.83). What I would drop is the "no password" patch, which I have not bothered to port.

**Comment by NiLuJe** [comment]
That's the other viable approach, true ;).

I... don't particularly care either way, we're icnreasingly going to be fileding issues either way (either about unsupported key formats; or about the lack of easy auth and/or sftp (which was also midly annoying to deal with)).

**Comment by Frenzie** [comment]
> What I would drop is the "no password" patch, which I have not bothered to port.

I wouldn't. I'd rather just drop it in that case. It works better than FTP, SCP, and Telnet, and the security aspect couldn't be less relevant.

The "but ewww..." in the OP I'd rephrase as "oh yes, yes, yes."

**Comment by anarcat** [comment]
not sure i can parse your comment there @Frenzie but i definitely mean "ewww" and by "ewww" i mean "passwords are evil, but password-less accounts are evil from an outer plane of existence that we shouldn't bring in this reality", if you want to rephrase. ;)

and if that's a patch, then, ewwwwwwwwwwwwwwwwwwwwww. :)

in any case, it looks like you fine folks have your hands full here... IMHO less patches sounds good (and i certainly don't mind having to use SSH keys!) but i'll let the maintainers decide what's best here.

**Comment by offset-torque** [comment]
@anarcat 
> password-less accounts are evil from an outer plane of existence that we shouldn't bring in this reality

- What is the threat model here in the KOReader context?
- What is the worst case scenario?

@Frenzie 
Should I put some warning in the guide related to passwordless login (which I also use since the beginning)?

**Comment by Frenzie** [comment]
@anarcat 
I have no problem with keys since obviously I use them plenty, but I'm not convinced the feature has much right to exist without allowing for passwordless login. SSH servers come a dime a dozen.


> Should I put some warning in the guide related to passwordless login (which I also use since the beginning)?

@offset-torque If there isn't any notice in there I suppose there better should be, but I think the menu phrasing `Login without password (DANGEROUS)` is fairly self-explanatory. Anybody who ignores that is clearly aware that any potential risk is negligible only if you run with wifi (or the server) always off except for a number of minutes here and there on a trusted local network.

**Comment by Frenzie** [comment]
@anarcat Also note that if we don't offer this people will use Telnet more.

**Comment by anarcat** [comment]
On 2024-04-21 01:41:20, Frans de Jonge wrote:
> @anarcat Also note that if we don't offer this people will use Telnet more.

I don't see how that follows; it's not like you *have* to provide a
passwordless telnet either.

**Comment by Frenzie** [comment]
Because the second sentence there has nothing to do with anything. ;-) We neither do nor have any intention to.
